OpenRA.Game/Actor.cs

```diff
@@ -96,6 +96,16 @@ internal Actor(World world, string name, TypeDictionary initDict)
 					throw new NotImplementedException("No rules definition for unit " + name);
 
 				Info = world.Map.Rules.Actors[name];
+
+				if (Info.HasTraitInfo<VariationsInfo>())
```

Comment:
Upstream we'd usually use `TraitInfoOrDefault` instead, e.g. like
```
var variantInfo = Info.TraitInfoOrDefault<VariationsInfo>();
if (variantInfo != null)
{
	[...]
		name = variantInfo.Variations...;
	[...]
}
```

Don't know how you want to handle that in AS/Engine.